lib/puppetserver/ca/puppet_config.rb

```diff
@@ -75,19 +74,28 @@ def resolve_settings(overrides = {})
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
-        settings[:cacert] = overrides.fetch(:cacert, '$cadir/ca_crt.pem')
-        settings[:cakey] = overrides.fetch(:cakey, '$cadir/ca_key.pem')
-        settings[:cacrl] = overrides.fetch(:cacrl, '$cadir/ca_crl.pem')
-        settings[:serial] = overrides.fetch(:serial, '$cadir/serial')
+        # Note the call to hostname below, (tbd from Justin's code)
+        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, `hostname`.chomp)
+
+        settings[:ca_name] =        overrides.fetch(:ca_name, 'Puppet CA: $certname')
+        settings[:root_ca_name] =   overrides.fetch(:root_ca_name, 'Puppet Root CA: $certname')
```

Comment:
I don't think the root CA should include the certname. The certname will most likely be the hostname and we expect users to be able to extract the root's cert and key and use them elsewhere on different hosts. We might want to put in a random value here - if we were using UUIDs for serials that'd be perfect - since we aren't yet we could just put in a seven digit hex? or something? I don't have a good answer, and could see using certname, I guess... What do you think @Magisus ?

Reply:
I'm a little curious how hard it would be to make the switch to using UUIDs as the serials. There are open tickets about it, and this could be our chance, but I don't really know what it would entail.

Reply:
In the meantime, a random number seems fine to me, if we think some uniqueness is needed. I don't really know what the CN is used for, in the context of the root.

Reply:
I can just as easily to a random 7 hex as a uuid.  Any preference?

Reply:
^^ Talked to Justin and doing 7 hex random